src/utils/path_publisher.cpp

```diff
@@ -0,0 +1,77 @@
+#include <ros/ros.h>
+#include <nav_msgs/Path.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <tf/transform_listener.h>
+#include <vector>
+
+
+// default parameters (these can be, probably, customized)
+
+#define DEFAULT_PATH_FRAME_ID "/map"
+#define DEFAULT_ROBOT_FRAME_ID "/base_link"
+#define DEFAULT_WORLD_FRAME_ID "/map"
+#define DEFAULT_TOPIC_NAME "/path"
+#define DEFAULT_NODE_NAME "path_publisher"
```

Comment:
All parameters should be configurable. An example of how to read parameters from a launch file can be found in `src/ros/init_utils.h`.